pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -220,18 +239,56 @@ func injectLibRequirements(pod *corev1.Pod, ctrName string, envKey string, envVa
 			continue
 		}
 
-		index := envIndex(ctr.Env, envKey)
-		if index < 0 {
-			pod.Spec.Containers[i].Env = append(pod.Spec.Containers[i].Env, corev1.EnvVar{
-				Name:  envKey,
-				Value: envVal(""),
-			})
-		} else {
-			if pod.Spec.Containers[i].Env[index].ValueFrom != nil {
-				return fmt.Errorf("%q is defined via ValueFrom", envKey)
+		err := setEnvVarWithFunc(&pod.Spec.Containers[i], envKey, envVal)
+		if err != nil {
+			return err
+		}
+
+		volumeAlreadyMounted := false
+		for _, vol := range pod.Spec.Containers[i].VolumeMounts {
+			if vol.Name == volumeName {
+				volumeAlreadyMounted = true
+				break
 			}
+		}
+		if !volumeAlreadyMounted {
+			pod.Spec.Containers[i].VolumeMounts = append(pod.Spec.Containers[i].VolumeMounts, corev1.VolumeMount{Name: volumeName, MountPath: mountPath})
+		}
+	}
+
+	return nil
+}
+
+// injectLibRequirements injects the minimal config requirements to enable instrumentation
+func injectLibRequirementsDotnet(pod *corev1.Pod, ctrName string) error {
+	for i, ctr := range pod.Spec.Containers {
+		if ctrName != "" && ctrName != ctr.Name {
+			continue
+		}
 
-			pod.Spec.Containers[i].Env[index].Value = envVal(pod.Spec.Containers[i].Env[index].Value)
+		err := setEnvVar(&pod.Spec.Containers[i], dotnetClrEnableProfilingKey, dotnetClrEnableProfilingValue)
+		if err != nil {
+			return err
+		}
+
+		err2 := setEnvVar(&pod.Spec.Containers[i], dotnetClrProfilerIdKey, dotnetClrProfilerIdValue)
```

Comment:
Creating new `errX` obj is not useful as we stop whenever we get one, re-using `err` is probably more appropriate.

Reply:
With the identity function you suggested, I removed the custom dotnet function entirely so this is no longer a concern. Done in 491425ab90ce7dfdc556249806debafecddab177